api/grpcserver/grpcserver_test.go

```diff
@@ -1426,8 +1427,39 @@ func TestMeshService(t *testing.T) {
 	}
 }
 
+func TestTransactionServiceSubmitUnsync(t *testing.T) {
+	require := require.New(t)
+	grpcService := NewTransactionService(&networkMock, txAPI, txMempool, &IsSyncerMock{})
+	shutDown := launchServer(t, grpcService)
+	defer shutDown()
+
+	// start a client
+	addr := "localhost:" + strconv.Itoa(cfg.NewGrpcServerPort)
+
+	// Set up a connection to the server.
+	conn, err := grpc.Dial(addr, grpc.WithInsecure())
+	require.NoError(err)
+
+	defer func() { require.NoError(conn.Close()) }()
+	c := pb.NewTransactionServiceClient(conn)
+
+	serializedTx, err := types.InterfaceToBytes(globalTx)
+	require.NoError(err, "error serializing tx")
+	res, err := c.SubmitTransaction(
+		context.Background(),
+		&pb.SubmitTransactionRequest{Transaction: serializedTx},
+	)
+	require.Error(err)
+	require.Nil(res)
+}
+
+type IsSyncerMock struct{ isSync bool }
```

Comment:
use existing `SyncerMock` from this test file

Reply:
I can't use SyncerMock without modifying it. I require a Syncer that can return true and false when `IsSynced` is called. You want me to modify the SyncerMock impl?

Reply:
You can modify it. Add a struct member called Synced, set to false by default, make IsSynced() return this value,
 then you can manually set it to true to control what it returns.

Reply:
I'll do it like this then.